### DISCUSSION TITLE
Date Time Format Specifications for danfo.to_datetime

### DISCUSSION THREAD

**Discussion by nico-hernandez** [opensource9ja/danfojs#156] [Q&A] [answered]
Hi lovely people!

Some few days ago I found this amazing library and I'ven learning a lot about it, but nowadays I'm facing issues that I couldn't figure out.
I'm parsing a JSON into Danfo DataFrame,  which contains a timestamp or datetime column in this format "YYYY-mm-ddTHH:MM:S.fZ" , where T is the separator for time and f is miliseconds.

I'd like to know **what should the format be for danfo.to_datetime parsing function?**  I've been trying many formats and the only thing that I get is "Invalid date, the date format not recognise".

Thanks in advance.

**Comment by steveoni** [comment]
@churchill1973 here are the format `"%Y-m-d%", "%m-d-Y%", "%m-d-Y H%M%S%"` specified here https://github.com/opensource9ja/danfojs/blob/3793a5d837687f0a99fdbd85ab0218d74331974a/danfojs-node/src/core/timeseries.js#L35

But i think the time format you specified above is a valid one but it's not included in the list of our time format.

You can give an example of such date format here and I can check if it will work by just adding it to the list

**Reply by nico-hernandez** [reply]
Thanks!

Indeed I could found that reference, but reading some other parts of the code I understood it was posible pass any other formats. Could you try one of these formats? 

1. "2021-03-19T08:25:35.000Z" (which is ISO 8601)
2. "2021-03-19T08:25:35.000"
3. "2021-03-19T08:25:35"

By the way, I know it's posible to create a datatime-like series from a column dataframe 
`let timestamp = dfd.to_datetime({data:dataset['Time'], format:format_string})` , but rather than creating a series, my use case it's simply changing the datatype of one column of the dataframe (let's say the dataframe has some other columns too). 
My final goal is filtering the dataframe by date using `Dataframe.query` or other filter function (I'm not sure if I could, join-merge is a workaround).

I tried something like this

1. `dataset['Time'] = dfd.to_datetime({data:dataset['Time'], format:format_string})`
2.  `dfd.to_datetime({data:dataset['Time'], format:format_string})`

None worked. How could I get it?

Thanks again!

**Comment by risenW** [comment, marked as answer]
@churchill1973 As of version 0.3.0 we now use JS default date internally. This means any Date string supported by JS now works, and you don’t have to specify a format. 

In case where you weird format not supported by the default JS Date. You would have to do some preprocessing.
